tests/MyGame/Example/Monster.go

```diff
@@ -6,9 +6,461 @@ import (
 	flatbuffers "github.com/google/flatbuffers/go"
 
 	MyGame "MyGame"
+	MyGame__Example2 "MyGame/Example2"
 )
 
 /// an example documentation comment: monster object
+type MonsterT struct {
+	Pos *Vec3T
+	Mana int16
+	Hp int16
+	Name string
+	Inventory []byte
+	Color Color
+	Test *AnyT
+	Test4 []*TestT
+	Testarrayofstring []string
+	Testarrayoftables []*MonsterT
+	Enemy *MonsterT
+	Testnestedflatbuffer []byte
+	Testempty *StatT
+	Testbool bool
+	Testhashs32Fnv1 int32
+	Testhashu32Fnv1 uint32
+	Testhashs64Fnv1 int64
+	Testhashu64Fnv1 uint64
+	Testhashs32Fnv1a int32
+	Testhashu32Fnv1a uint32
+	Testhashs64Fnv1a int64
+	Testhashu64Fnv1a uint64
+	Testarrayofbools []bool
+	Testf float32
+	Testf2 float32
+	Testf3 float32
+	Testarrayofstring2 []string
+	Testarrayofsortedstruct []*AbilityT
+	Flex []byte
+	Test5 []*TestT
+	VectorOfLongs []int64
+	VectorOfDoubles []float64
+	ParentNamespaceTest *MyGame.InParentNamespaceT
+	VectorOfReferrables []*ReferrableT
+	SingleWeakReference uint64
+	VectorOfWeakReferences []uint64
+	VectorOfStrongReferrables []*ReferrableT
+	CoOwningReference uint64
+	VectorOfCoOwningReferences []uint64
+	NonOwningReference uint64
+	VectorOfNonOwningReferences []uint64
+	AnyUnique *AnyUniqueAliasesT
+	AnyAmbiguous *AnyAmbiguousAliasesT
+	VectorOfEnums []Color
+}
+
+func MonsterPack(builder *flatbuffers.Builder, t *MonsterT) flatbuffers.UOffsetT {
+	if t == nil { return 0 }
+	nameOffset := builder.CreateString(t.Name)
+	inventoryOffset := flatbuffers.UOffsetT(0)
+	if t.Inventory != nil {
+		inventoryOffset = builder.CreateByteString(t.Inventory)
+	}
+	testOffset := flatbuffers.UOffsetT(0)
+	if t.Test != nil {
+		switch t.Test.Type {
```

Comment:
Would it be worth factoring this switch out into a `Pack` method on the union (and same for `UnPack`)? That way you reduce code bloat if a union is used more than once?

Reply:
`Pack` usage is worth and very straightforward.
`UnPack` is a little bit more difficult.
We can achieve it using something similar

```go
func AnyUnPack(t Any, f func(*flatbuffers.Table) bool) *AnyT {
	switch t {
	case AnyMonster:
		x := Monster{}
		if f(&x._tab) {
			return &AnyT{ Type: AnyMonster, Value: x.UnPack() }
		}
	case AnyTestSimpleTableWithEnum:
		x := TestSimpleTableWithEnum{}
		if f(&x._tab) {
			return &AnyT{ Type: AnyTestSimpleTableWithEnum, Value: x.UnPack() }
		}
	case AnyMyGame_Example2_Monster:
		x := Monster{}
		if f(&x._tab) {
			return &AnyT{ Type: AnyMyGame_Example2_Monster, Value: x.UnPack() }
		}
	}
	return nil
}
...
t.Test = AnyUnPack(rcv.TestType(), rcv.Test)
```

Any comments? 
The API is not clean because it depends on the field name that may change for different Union usages.

Reply:
I don't think I understand what this code does. Why is the `func` necessary? What does `UnPack()` do on what looks like an empty object? I am probably missing some Go intricacy. @iceb0y ?

This is what it looks like in C++:
https://github.com/google/flatbuffers/blob/cfb4ecf6f06ffc4b93a70b85bc66903471e1dc8b/tests/monster_test_generated.h#L2820-L2836

Reply:
The `func` is not necessary. We can do something like this:

```
testTable := flatbuffers.Table{}
if rcv.Test(&testTable) {
  t.Test = AnyUnPack(testTable, rcv.TestType())
}
```

and

```
func AnyUnPack(table flatbuffers.Table, type Any) *AnyT {
  ...
}
```

Reply:
Sorry, the func part was my fault. I thought that to return the union table first I must guarantee that it was defined, That's the reason why I sent the function as a function argument but I checked the code and saw that I can grab the table before entering the function.
`func AnyUnPack(t Any, table flatbuffers.Table) *AnyT {`

I didn't understood what you mean by 
>  What does UnPack() do on what looks like an empty object?